subprojects/plugins/src/main/java/org/gradle/api/tasks/bundling/War.java

```diff
@@ -68,6 +73,9 @@ public War() {
         renameSpec.into("");
         renameSpec.from((Callable<File>) War.this::getWebXml);
         renameSpec.appendCachingSafeCopyAction(new RenamingCopyAction(Transformers.constant("web.xml")));
+
+        ObjectFactory objectFactory = getProject().getObjects();
```

Comment:
Add a `@Inject ObjectFactory getObjectFactory() { throw ... }` getter instead of reaching into a project.